src/restrictededitingmodeediting.js

```diff
@@ -0,0 +1,275 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module restricted-editing/restrictededitingmodeediting
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import RestrictedEditingNavigationCommand from './restrictededitingmodenavigationcommand';
+import {
+	extendMarkerOnTypingPostFixer,
+	resurrectCollapsedMarkerPostFixer,
+	setupExceptionHighlighting,
+	upcastHighlightToMarker
+} from './restrictededitingmode/converters';
+import { getMarkerAtPosition, isSelectionInMarker } from './restrictededitingmode/utils';
+
+/**
+ * The Restricted Editing Mode editing feature.
+ *
+ * * It introduces the exception marker group that renders to `<spans>` with the `ck-restricted-editing-exception` CSS class.
+ * * It registers the `'goToPreviousRestrictedEditingRegion'` and `'goToNextRestrictedEditingRegion'` commands.
+ * * Also enables highlighting exception markers that are selected.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class RestrictedEditingModeEditing extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'RestrictedEditingModeEditing';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor( editor ) {
+		super( editor );
+
+		editor.config.define( 'restrictedEditing', {
+			allowedCommands: [ 'bold', 'italic', 'link' ]
+		} );
+
+		/**
+		 * Command names that are enabled outside non-restricted regions.
+		 *
+		 * @type {Set<string>}
+		 * @private
+		 */
+		this._alwaysEnabled = new Set( [ 'undo', 'redo', 'goToPreviousRestrictedEditingRegion', 'goToNextRestrictedEditingRegion' ] );
+
+		/**
+		 * Commands allowed in non-restricted areas.
+		 *
+		 * Commands always enabled combines typing feature commands: `'typing'`, `'delete'` and `'forwardDelete'` with commands defined
+		 * in the feature configuration.
+		 *
+		 * @type {Set<string>}
+		 * @private
+		 */
+		this._allowedInException = new Set( [ 'input', 'delete', 'forwardDelete' ] );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+
+		const allowedCommands = editor.config.get( 'restrictedEditing.allowedCommands' );
+
+		allowedCommands.forEach( commandName => this._allowedInException.add( commandName ) );
+
+		this._setupConversion();
+		this._setupCommandsToggling();
+
+		// Commands & keystrokes that allow navigation in the content.
+		editor.commands.add( 'goToPreviousRestrictedEditingRegion', new RestrictedEditingNavigationCommand( editor, 'backward' ) );
```

Comment:
`goToPreviousRestrictedEditingException`

You use the `restricted-editing-exception` name in other places already and it should be fine.